common/errors.go

```diff
@@ -0,0 +1,10 @@
+package common
+
+// Errors that don't present security issues when returned to students.
+type SecureError struct {
```

Comment:
I'm not a fan of these SecureErrors.
The concept makes sense, but I am worried about the design/implementation.

 - Are there endpoints aside from submit that need to worry about these?
 - What about api/core, does it need to know about these?
 - What about logging? We were thinking about students, but if we are thinking about sensitive information we should also think about what goes into the logs (currently no sensitive information is allowed to go in them).
 - Where is the safe place to output sensitive information?
I think doing this right requires us to go deeper.

We may also need to go in the opposite directions.
A type like this implies that all other errors are insecure, but we have not had this assumption up until now.
I think we first need to establish where error information can end up:
 - stdout/stderr
 - logs
 - api responses

Then, we can have something like "ConfidentialErrors" whose Error() method will output something safe.
But will know how to be interpreted by the `api/error` and `log` packages.
So anyone calling Error() will get something safe and the places where these errors may end up know how to better handle things.

These are my ideas, but this requires more thought.
Take a ponder on it.